crates/biome_js_analyze/src/lint/suspicious/no_alert.rs

```diff
@@ -0,0 +1,126 @@
+use crate::services::semantic::Semantic;
+use biome_analyze::{Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsExpression, JsCallExpression, global_identifier};
+use biome_rowan::AstNode;
+
+declare_lint_rule! {
+    /// Disallow the use of `alert`, `confirm`, and `prompt`.
+    ///
+    /// JavaScript's `alert`, `confirm`, and `prompt` functions are widely considered to be obtrusive
+    /// as UI elements and should be replaced by a more appropriate custom UI implementation.
+    /// Furthermore, `alert` is often used while debugging code, which should be removed before
+    /// deployment to production.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// alert("here!");
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// confirm("Are you sure?");
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// prompt("What's your name?", "John Doe");
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// customAlert("Something happened!");
+    /// ```
+    ///
+    /// ```js
+    /// customConfirm("Are you sure?");
+    /// ```
+    ///
+    /// ```js
+    /// customPrompt("Who are you?");
+    /// ```
+    ///
+    /// ```js
+    /// function foo() {
+    ///     const alert = myCustomLib.customAlert;
+    ///     alert();
+    /// }
+    /// ```
+    pub NoAlert {
+        version: "1.0.0",
+        name: "noAlert",
+        language: "js",
+        sources: &[RuleSource::Eslint("no-alert")],
+        recommended: true,
```

Comment:
We usually don't recommend new rules too.

```suggestion
        recommended: false,
```